### ISSUE TITLE
[REVIEW]: CoralP: Flexible visualization of the human phosphatome

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1837]
**Submitting author:** @dphansti (<a href="http://orcid.org/0000-0003-2123-0051">Douglas Phanstiel</a>)
**Repository:** <a href="https://github.com/PhanstielLab/coralp" target ="_blank">https://github.com/PhanstielLab/coralp</a>
**Version:** v1.0.1
**Editor:** @majensen
**Reviewer:** @daissi, @stulacy
**Archive:** <a href="https://doi.org/10.5281/zenodo.3560845" target="_blank">10.5281/zenodo.3560845</a>

## Status

[![status](https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65/status.svg)](https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65"><img src="https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65/status.svg)](https://joss.theoj.org/papers/3fd2226fa440647b187de756beb21c65)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/PhanstielLab/coralp" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@daissi & @stulacy, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @majensen know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PhanstielLab/coralp">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@dphansti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PhanstielLab/coralp">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@dphansti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @daissi, @stulacy it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01837/joss.01837/10.21105.joss.01837.pdf)

**Comment by stulacy** [comment]
Hi!

Could you please clean up some of the typos in the paper, such as: 'inlcuding', the odd symbol in the first sentence of Methods, and capitalising 'JavaScript'.
Also, the first two sentences in Methods are somewhat duplicated.

**Comment by dphansti** [comment]
@stulacy Thanks for finding these issues. We have corrected those typos and made several other tiny edits to improve clarity and accuracy.

**Comment by daissi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01837/joss.01837/10.21105.joss.01837.pdf)

**Comment by majensen** [comment]
@daissi, @stulacy  - how are the checks coming? Can I ask you to check off or create issues for @dphansti for these over the next week? Let me know a timeline that works for you.

**Comment by daissi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01837/joss.01837/10.21105.joss.01837.pdf)

**Comment by daissi** [comment]
> @daissi, @stulacy - how are the checks coming? Can I ask you to check off or create issues for @dphansti for these over the next week? Let me know a timeline that works for you.

Everything is fine from my side! :thumbsup:

**Comment by stulacy** [comment]
The only thing left I need to do is try and run the package locally which I
will try and complete tonight.

The only thing I'm unsure about is how to assess the "Automated Tests"
point, since this is a web-app rather than a package interfaced with via an
API. Is this criteria not valid for web-apps or should they still have some
sort of testing procedure?

**Comment by stulacy** [comment]
I've gone through the remaining points and am happy that the submission meets them all, although I still am unsure how to handle the Automated Tests criteria as mentioned above. Any thoughts @majensen ?

**Comment by majensen** [comment]
Thanks @stulacy - I will have a look and make a rec within a couple days. (Sorry I have been slow to respond for a couple weeks)

**Comment by majensen** [comment]
@dphansti  - per @stulacy comments: I agree that there should be some tests associated with the R code (the JS is less critical IMO). I wish I had time to help you stub these out, but for now let me suggest you look at https://testthat.r-lib.org/ - this looks really easy to use, and is a common testing system in other R-based pubs in JOSS. Here's an example : https://github.com/junbinzhao/FluxCalR/tree/master/tests
I can provide more advice if needed - but I wanted to get you looking at this. 
thanks!

**Comment by dphansti** [comment]
@daissi, @stulacy, @majensen We have added unit tests using the `testthat` package for key functions, as well as an application-level test using the `shinytest` package that emulates simple tasks in the user interface and compares with pre-validated results. We also configured Travis CI to automatically run these tests with every change and report failures.

**Comment by stulacy** [comment]
Great, thanks @dphansti . I've completed my review.

**Comment by majensen** [comment]
@dphansti thanks very much for that additional work. Thanks @stulacy and @daissi for the review!
Next step is to create a public archive of the software with a DOI. @dphansti, you can use [Zenodo](zenodo.org) for this purpose, or other service you prefer. If you're unfamiliar with this process, have a look at https://github.com/openjournals/joss-reviews/issues/1839#issuecomment-560048729 for a brief tutorial. Once this is done, please report the DOI back here.
thanks again

**Comment by dphansti** [comment]
@majensen Done.  Here is the doi 10.5281/zenodo.3560845

**Comment by majensen** [comment]
@whedon set 10.5281/zenodo.3560845 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3560845" target="_blank">10.5281/zenodo.3560845</a> is the archive.

**Comment by majensen** [comment]
@whedon set v1.0.1 as version

**Comment by whedon** [comment]
OK. v1.0.1 is the version.

**Comment by majensen** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01837/joss.01837/10.21105.joss.01837.pdf)

**Comment by majensen** [comment]
@openjournals/joss-eics this paper is recommended for publication. Please initiate final checks- thanks!

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.febslet.2012.05.008 is OK
- 10.1109/TVCG.2011.185 is OK
- 10.1038/nrd773 is OK
- 10.7717/peerj.126 is OK
- 10.1042/BJ20061548 is OK
- 10.1086/422827 is OK
- 10.1126/scisignal.aag1796 is OK
- 10.1038/msb.2010.52 is OK
- 10.1111/febs.12000 is OK
- 10.1074/jbc.M505159200 is OK
- 10.1016/j.cels.2018.07.001 is OK
- 10.1111/j.1742-4658.2008.06250.x is OK
- 10.1096/fj.02-1212rev is OK
- 10.1038/nrc1837 is OK
- 10.1186/s12859-016-1433-7 is OK
- 10.1126/science.1075762 is OK
- 10.1038/nrm2039 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1148

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1148, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1201965436428079105) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1149
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01837](https://doi.org/10.21105/joss.01837)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @daissi & @stulacy for reviewing, and @majensen for editing, and congratulations to  @dphansti and co-authors!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01837/status.svg)](https://doi.org/10.21105/joss.01837)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01837">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01837/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01837/status.svg
   :target: https://doi.org/10.21105/joss.01837
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01837/status.svg)](https://doi.org/10.21105/joss.01837)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
